lib/core/src/irods_service_account.cpp

```diff
@@ -0,0 +1,132 @@
+            ret = parse_escaped_kvp_string(cfg_line, service_acct_cfg);
+            if (!ret.ok()) {
+                irods::log(LOG_WARNING,
+                           fmt::format("{} - {} parse failure on line {}",
+                                       __PRETTY_FUNCTION__,
+                                       SERVICE_ACCOUNT_CONFIG_FILE,
+                                       cfg_line_no));
+            }
+        }
+
+        const auto& svc_acct_itr = service_acct_cfg.find(SERTICE_ACCOUNT_KW);
+        if (svc_acct_itr == service_acct_cfg.end()) {
+            THROW(
+                KEY_NOT_FOUND, fmt::format("could not find {} in {}", SERTICE_ACCOUNT_KW, SERVICE_ACCOUNT_CONFIG_FILE));
+        }
+        service_account_username = svc_acct_itr->second;
+
+        // compare uids rather than usernames, to account for overlapping uids
+        // NOTE: do not reference the strings inside these structs, as their buffer goes out of scope
+        struct passwd svc_acct_pwd{};
+        struct passwd* svc_acct_pwdp = nullptr;
+        const auto _pwd_init_len = sysconf(_SC_GETPW_R_SIZE_MAX);
+        size_t pwd_buf_len = PWD_BUF_LEN_DEFAULT; // a reasonable buffer size in case sysconf gives us something awful
+        if (_pwd_init_len >= PWD_BUF_LEN_MINIMUM) {
+            pwd_buf_len = static_cast<size_t>(_pwd_init_len);
+        }
+        int pwd_e = 0;
+        while (true) {
+            // this sure is a lot of fuss to allocate a buffer for data we're not going to use anyway...
+            const std::unique_ptr<char[]> pwd_buf( // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+                new char[pwd_buf_len]);
+            pwd_e =
+                getpwnam_r(service_account_username.c_str(), &svc_acct_pwd, pwd_buf.get(), pwd_buf_len, &svc_acct_pwdp);
+            if (pwd_e != ERANGE) {
+                break;
+            }
+            // buffer wasn't big enough, try again
+            const size_t new_len = pwd_buf_len * 2;
+            // check for overflow
+            // FIXME: compilers can optimize away this check
+            if (pwd_buf_len > new_len) {
+                THROW(SYS_MALLOC_ERR, "integer overflow when allocating space for passwd struct data");
+            }
+            pwd_buf_len = new_len;
+        }
+        switch (pwd_e) {
+            case 0:
+                break;
+            case EMFILE:
+            case ENFILE:
+                THROW(SYS_OUT_OF_FILE_DESC - pwd_e, "hit file descriptor limit");
+                break;
+            case EINTR:
+                THROW(SYS_CAUGHT_SIGNAL - pwd_e, "caught signal");
+                break;
+            default:
+                THROW(SYS_UNKNOWN_ERROR - pwd_e, "unknown error when calling getpwnam_r");
+                break;
+        }
+        if (svc_acct_pwdp == nullptr) {
+            irods::log(LOG_WARNING, fmt::format("%s - could not find user with name %s", service_account_username));
```

Comment:
Why not use the new logger?

Reply:
Marking this one as resolved since discussion is happening in the other one